Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -1392,6 +1394,28 @@ namespace AZ
                 AZ_PROFILE_SCOPE(AzCore, "ComponentApplication::Tick:OnTick");
                 EBUS_EVENT(TickBus, OnTick, m_deltaTime, ScriptTimePoint(now));
             }
+
+            // If tick rate limiting is on, ensure (1 / g_simulation_tick_rate) ms has elapsed since the last frame,
+            // sleeping if there's still time remaining.
+            if (g_simulation_tick_rate > 0.f)
+            {
+                now = AZStd::chrono::system_clock::now();
+
+                // Work in microsecond durations here as that's the native measurement time for time_point
+                constexpr float microsecondsPerSecond = 1000.f * 1000.f;
+                const AZStd::chrono::microseconds timeBudgetPerTick(static_cast<int>(microsecondsPerSecond / g_simulation_tick_rate));
+                AZStd::chrono::microseconds timeUntilNextTick = m_currentTime + timeBudgetPerTick - now;
+
+                if (timeUntilNextTick.count() > 0)
+                {
+                    AZStd::this_thread::sleep_for(timeUntilNextTick);
+                    m_currentTime = AZStd::chrono::system_clock::now();
```

Comment:
Isn't this erasing time? This appears to lose all time you might have slept for so that our deltaTimes will now be contracted and incorrect on the next tick. Also just due to how frequently modern schedulers operate, the sleep here will not sleep microseconds even if we request so. Unless there are no other high priority tasks pending this will likely end up sleeping for 10 milliseconds or more.

Reply:
I don't think this will play nice with the timing system.
In the normal run of Editor or Standalone ```deltaOverride```  is set from the CryTimer by the caller to this function.
So the deltaTime will be set to that delta(there is a bunch of logic in CryTimer and what it does), so ```m_currentTime``` is not used to calculate the delta passed to the OnTick call.

There are also a bunch of other places with the CryTimer is used to get the delta time, which will include this additional sleep (example: TrackView + The movie system do this).

I have a change i'm working on currently to unify and clean up how this 'time' is managed and calculated. i think it might be better and simpler to add ```g_simulation_tick_rate``` with that change.

Reply:
I shouldn't actually be updating `m_currentTime` at all, that's an artifact from me moving the logic over from somewhere else that wasn't already storing the last timestamp. Fun fact: the reason nothing broke in my testing is because of the exact reason Sean outlined, the `deltaOverride` parameter is always set by the Cry Timer at the moment, so I never had a chance to steal time from the delta. It also appears to be correctly.

As for specifying the delta at microsecond resolution, that's a fair point - my thought process was that I'd give as accurate as possible timing information and sleep for a best effort duration, as truncation down to milliseconds is not a big deal; if preferred, I can switch the timing info to milliseconds (or tear this out and wait for Sean's change).